python/shotgun_desktop/initialization/shotgun.py

```diff
@@ -123,16 +124,17 @@ def get_app_store_credentials(connection, proxy):
 
     # connect to the app store
     try:
-        sg_app_store = Shotgun(constants.SGTK_APP_STORE, script, key, http_proxy=proxy)
+        sg_app_store = Shotgun(constants.SGTK_APP_STORE, script, key, http_proxy=proxy, connect=False)
 
         # pull down the full script information
         app_store_script = sg_app_store.find_one(
             "ApiUser",
             [["firstname", "is", script]],
             fields=["type", "firstname", "id", "salted_password"])
-    except:
```

Comment:
Do not swallow exceptions anymore, this was the source of the confusion. Handle what we understand and let the rest flow.